adding powerplatform_rest_query and powerplatform_rest (#347)
microsoft/terraform-provider-power-platform: docs/resources/rest.md

---
# generated by https://github.com/hashicorp/terraform-plugin-docs
page_title: "powerplatform_rest Resource - powerplatform"
subcategory: ""
description: |-
  Resource to execute web api requests
---

# powerplatform_rest (Resource)

Resource to execute web api requests

## Example Usage

```terraform
terraform {
  required_providers {
    powerplatform = {
      source = "microsoft/power-platform"
    }
  }
}

provider "powerplatform" {
  use_cli = true
}

resource "powerplatform_environment" "env" {
  display_name     = "sample_data_environment"
  location         = "europe"
  environment_type = "Sandbox"
  dataverse = {
    language_code     = "1033"
    currency_code     = "USD"
    security_group_id = "00000000-0000-0000-0000-000000000000"
  }
}

resource "powerplatform_rest" "install_sample_data" {

>   environment_id = powerplatform_environment.env.id

Author: fixed, userdocs where not updated

---
# generated by https://github.com/hashicorp/terraform-plugin-docs
page_title: "powerplatform_rest Resource - powerplatform"
subcategory: ""
description: |-

>   Resource to execute web api requests. There are four distinct operations, that you can define idepenetly. The HTTP response' body of the operation, that was called as last, will be returned in 'output.body' \n\n:

Reviewer: ```suggestion
  Resource to execute web api requests. There are four distinct operations, that you can define independently. The HTTP response' body of the operation, that was called as last, will be returned in 'output.body' \n\n:
```

  * Create: will be called once during the lifecycle of the resource (first 'terraform apply')
  * Read: terraform will call this operation every time during 'plan' and 'apply' to get the current state of the resource
  * Update: will be called every time during 'terraform apply' if the resource has changed (change done by the user or different values returned by the 'read' operation than those in the current state)
  * Destroy: will be called once during the lifecycle of the resource (last 'terraform destroy')

>   \n\nYOu don't have to define all the operations but there are some things to consider:

Reviewer: ```suggestion
  \n\nYou don't have to define all the operations but there are some things to consider:
```

  * lack of 'create' operation will result in no reasource being created. If you only need to read values consider using datasource 'powerplatform_rest_query' instead
  * lack of 'read' operation will result in no resource changes being tracked. That means that the 'update' operation will never be called

>   * lack of destroy will couse, that the resource will not be deleted during 'terraform destroy'

Reviewer: ```suggestion
  * lack of destroy will cause that the resource to not be deleted during 'terraform destroy'
```
